src/App.tsx

```diff
@@ -42,54 +43,104 @@ export function getReorderedGoods(
 }
 
 // DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
+};
+
+export class App extends React.Component<{}, State> {
+  state = { isReversed: false, sortType: SortType.NONE }
+
+  sortAlphabetically = () => {
+    this.setState({ sortType: SortType.ALPHABET });
+  }
+
+  sortByLength = () => {
+    this.setState({ sortType: SortType.LENGTH });
+  }
+
+  reverseGoods = () => {
+    this.setState((prev) => ({ isReversed: !prev.isReversed }));
+  }
+
+  reset = () => {
+    this.setState({ isReversed: false, sortType: SortType.NONE });
+  }
+
+  render() {
+    const { isReversed, sortType } = this.state;
+
+    const visibleGoods = getReorderedGoods(goodsFromServer, {
+      isReversed,
+      sortType,
+    });
+
+    if (sortType === SortType.ALPHABET) {
+      visibleGoods.sort((a, b) => a.localeCompare(b));
+    } else if (sortType === SortType.LENGTH) {
+      visibleGoods.sort((a, b) => a.length - b.length);
+    }
+
+    if (isReversed) {
+      visibleGoods.reverse();
+    }
```

Comment:
I think you should do all the sorting and reversing inside of getReorderedGoods function. That's why we are passing there the sortType and isReversed as arguments